test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -309,21 +323,273 @@ describe('Poco', async () => {
-            expect(await iexecPoco.balanceOf(sponsor.address)).to.be.equal(0);
-            expect(await iexecPoco.frozenOf(sponsor.address)).to.be.equal(0);
+            await expect(
+                iexecPocoAsScheduler.finalize(taskId, results, '0x'),
+            ).to.changeTokenBalances(
+                iexecPoco,
+                [requester, sponsor, appProvider, datasetProvider],
+                [
+                    0, // requester balance is unchanged, only frozen is changed
+                    0,
+                    appPrice, // app provider is rewarded
+                    0, // but dataset provider is not rewarded
+                ],
+            );
+            expect(await iexecPoco.frozenOf(requester.address)).to.be.equal(
+                requesterFrozenBefore - taskPrice,
+            );
+            expect(await iexecPoco.frozenOf(sponsor.address)).to.be.equal(sponsorFrozenBefore);
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.resultsCallback).to.equal('0x'); // deal without callback
         });
     });
 
-    // TODO: Continue `finalize` tests migration (`Should/Should not`)
+    it('Should finalize task after reveal deadline with at least one reveal', async () => {
+        const volume = 1;
+        const { orders } = buildOrders({
+            assets: ordersAssets,
+            requester: requester.address,
+            prices: ordersPrices,
+            volume,
+            trust: 3,
+        });
+        const { dealId, taskId, taskIndex } = await iexecWrapper.signAndMatchOrders(orders);
+        await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+        const workerTaskStake = await iexecPoco
+            .viewDeal(dealId)
+            .then((deal) => deal.workerStake.toNumber());
+        const workers = [worker1, worker2];
+        for (const worker of workers) {
+            const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                taskId,
+                resultDigest,
+                worker,
+            );
+            const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+                worker.address,
+                taskId,
+                emptyEnclaveAddress,
+                scheduler,
+            );
+            await iexecWrapper.depositInIexecAccount(worker, workerTaskStake);
+            await iexecPoco
+                .connect(worker)
+                .contribute(
+                    taskId,
+                    resultHash,
+                    resultSeal,
+                    emptyEnclaveAddress,
+                    emptyEnclaveSignature,
+                    schedulerSignature,
+                )
+                .then((tx) => tx.wait());
+        }
+        await iexecPoco
+            .connect(worker1)
+            .reveal(taskId, resultDigest)
+            .then((tx) => tx.wait());
+        const taskBefore = await iexecPoco.viewTask(taskId);
```

Comment:
Why name it task Before

Reply:
Thanks updated [Rename variable to task](https://github.com/iExecBlockchainComputing/PoCo/pull/117/commits/ab38da3ab35fbd105d311bc20233b25221b9d994) !